src/createEventEmitterProxy.ts

```diff
@@ -93,8 +128,43 @@ export function createEventEmitterProxy<Type extends EventEmitterLike>(
       }
 
       const value = target[name];
-      if (value instanceof Function) {
+
+      if (typeof value === 'function') {
         return function (this: unknown, ...args: any[]) {
+          const unwrappedHandler = args[1];
+          if (name === 'once') {
+            const wrappedHandler = (...handlerArgs: any[]) => {
+              removeEvent(args[0], wrappedHandler);
+              return unwrappedHandler(...handlerArgs);
+            };
+            args[1] = wrappedHandler;
+          }
+          if (
+            name === 'on' ||
+            name === 'addListener' ||
+            name === 'prependListener' ||
+            name === 'once'
+          ) {
+            eventsAdded.push({
+              addedWith: name,
+              name: args[0],
+              unwrappedHandler,
+              handler: args[1],
+              filtered: !eventFilter(args[0]),
+            });
+          } else if (name === 'off' || name === 'removeListener') {
+            const eventAdded = eventsAdded.find(
+              ({ name: addedName, unwrappedHandler: addedUnwrappedHandler }) =>
+                addedName === args[0] && addedUnwrappedHandler === args[1],
+            );
+            // this should never really happen unless you've called removeListener for something that is already not there.
+            if (eventAdded === undefined) {
+              return target;
```

Comment:
HJmm, this doesn't seem quite right. If `removeListener` is called with an event that we don't know about, I think we should pass it on down through to the `value.apply` call below.

Returning `target` here is a bit strange because `target` is the underlying event emitter, but this function is the getter. So it means that something like `proxy.off('foo', handler)` would get the event emitter for `off`, making it `eventEmitter('foo', handler)`, but `eventEmitter` isn't a function, so that's not right.

Reply:
Oh I'm sorry, we're within the returned function, not within the `getter`. My bad, this is not doing what I thought.

Reply:
I see, and we'd want to avoid passing it down just in case that exact same event and listener was added via another proxy. That makes sense.

Reply:
Hmm yeah I couldnt decide on if it made more sense to just send the removeListner down or not. But I think you are probably right, sending it down makes more sense. In the event you are removing a listener that was not added via the proxy, but is present on the event emitter, the question was 'should we still remove it?' I talked myself in-and-out of it a few times over... But yeah I think you are right, we should send it.

As for the return value, This return is not the getter, it is essentially a wrapped version of the function returned by the getter. As such, the expected return value after calling `eventEmitter.removeListener('abc', () => 123)` is the eventEmitter itself. To be more specific to your example, the getter which returns a function for  `proxy.off` is still intact - we return a function. Its the return value of this function that we are changing.

Reply:
oaky updated, take a look!

Reply:
Sorry for the additional mixup here! See my follow-up two comments, I think the way you have it here is actually safer in the case where two proxies point to the same target.

e.g. Lets say that `a` proxies `target`, and `b` proxies `target`.

If I run this code:

```
const handleNetworkChanged = () => console.log('network changed');
a.on('networkChanged', handleNetworkChanged)

b.on('networkChanged', handleNetworkChanged)

a.off('networkChanged', handleNetworkChanged)
a.off('networkChanged', handleNetworkChanged) // duplicate call to `off` here

target.emit('networkChanged')
```

Then we'd want `handleNetworkChanged` to still be called because `b` registered it and never unregistered it.

So your code is correct here, but if we passed it through like I suggested originally, this case would be broken

Reply:
Maybe that case would make for a good test as well, that's a pretty subtle scenario. Doesn't seem that unlikely either. I can see that happening if the thing we want to trigger is not directly related to either event emitter.

e.g. what if we want to trigger saving state if a network status changes for an "active" network, that might show up as this kinda scenario, where different proxies trigger the same persist call.

Reply:
Ive reverted the last two commits. If its okay with you, maybe we can create an issue to create test cases for 'many-to-one proxies to eventEmitter' - I suspect there might be a few nuanced items to figure out there.

Reply:
Sure, sounds good